Core/include/Acts/Geometry/detail/PortalGenerators.hpp

```diff
@@ -0,0 +1,97 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/Portal.hpp"
+#include "Acts/Geometry/VolumeBounds.hpp"
+#include "Acts/Geometry/detail/DetectorVolumeUpdators.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+
+#include <exception>
+#include <memory>
+#include <tuple>
+#include <vector>
+
+namespace Acts {
+namespace Experimental {
+namespace detail {
+
+/// @brief Generator function for creation of portal surfaces
+/// for a cylindrical volume
+///
+/// @param dTransform a contextually resolved transform
+/// @param dBounds the detecor volume bounds
+/// @param dVolume the reference to the detector volume which generates this volume
+///
+/// @return a vector of newly created portals with registered inside volume
+inline static std::vector<std::shared_ptr<Portal>> portals(
```

Comment:
That's a fairly generic function name. Can this be more explicit?

Reply:
Updated.